src/freertos_drivers/esp_idf/EspIdfWiFi.cxx

```diff
@@ -0,0 +1,1367 @@
+        LOG(VERBOSE, "wifi: STA unregistered on mDNS.");
+    }
+#endif
+}
+
+//
+// EspIdfWiFiBase::try_restore_sta()
+//
+void EspIdfWiFiBase::mdns_restore_sta()
+{
+#if !defined(CONFIG_MDNS_PREDEF_NETIF_STA)
+    OSMutexLock locker(&lock_);
+    HASSERT(mdnsStaLockCount_ != 0);
+    if (--mdnsStaLockCount_ == 0)
+    {
+        if (staIface_)
+        {
+            // If the STA interface is not registered by default, register it.
+            ESP_ERROR_CHECK(mdns_register_netif(staIface_));
+            mdns_event_actions_t action = static_cast<mdns_event_actions_t>(
+                MDNS_EVENT_ENABLE_IP4 | MDNS_EVENT_ENABLE_IP6 |
+                MDNS_EVENT_ANNOUNCE_IP4 | MDNS_EVENT_ANNOUNCE_IP6);
+            ESP_ERROR_CHECK(mdns_netif_action(staIface_, action));
+            LOG(VERBOSE, "wifi: STA registered on mDNS.");
+        }
+    }
+#endif
+}
+
+//
+// EspIdfWiFiBase::mdns_scanning_start_or_trigger_refresh()
+//
+void EspIdfWiFiBase::mdns_scanning_start_or_trigger_refresh()
+{
+    if (!mdnsClientStarted_)
+    {
+        // Start the client if not already started.
+        mdnsClientStarted_ = true;
+        notify();
+        LOG(INFO, "wifi: mdns_scan() start.");
+    }
+    else if (!mdnsClientTrigRefresh_)
+    {
+        // Trigger the mDNS state machine to execute early. If a query is
+        // already taking place, a new one will start immediately following it.
+        // If a query it not taking place, one will be started immediately.
+        mdnsClientTrigRefresh_ = true;
+        CallbackExecutable *e =
+            new CallbackExecutable([this](){this->timer_.ensure_triggered();});
+        this->service()->executor()->add(e);
+        LOG(INFO, "wifi: mdns_scan() trigger.");
+    }
+}
+
+//
+// EspIdfWiFiBase::wifi_event_handler()
+//
+void EspIdfWiFiBase::wifi_event_handler(
+    esp_event_base_t base, int32_t id, void *data)
+{
+    HASSERT(base == WIFI_EVENT);
+
+    switch (id)
+    {
+        default:
+            LOG(INFO, "wifi: wifi_event_handler() unknown id: %i", (int)id);
+            break;
+        case WIFI_EVENT_WIFI_READY:
+            started_ = true;
+            LOG(INFO, "wifi: WiFi started.");
+            break;
+        case WIFI_EVENT_STA_START:
+        {
+            std::string ssid;
+            std::string pass;
+            uint8_t authmode;
+            uint8_t channel;
+            last_sta_get(&ssid, &pass, &authmode, &channel);
+            if (ssid.empty())
+            {
+                // No valid "fast" connect parameters.
+                esp_wifi_scan_start(&SCAN_CONFIG, false);
+                LOG(INFO, "wifi: STA start, scanning...");
+            }
+            else
+            {
+                // Try the fast connect parameters first.
+                sta_connect(ssid, pass, authmode, channel);
+#if (LOGLEVEL == VERBOSE)
+                LOG(VERBOSE,
+                    "wifi: STA start, ssid: %s, pass: %s, connecting...",
```

Comment:
I'm not excited about putting the password to the log, even if it is guarded by verbosity. Maybe put only the length?

Reply:
Let's just remove the password printing.